skedge-backend/database/services/database-service.js

```diff
@@ -100,33 +101,62 @@ module.exports = {
     },
 
     insertOneInDatabase: function (objectJSON, collectionName) {
+      return new Promise ((resolve, reject) => {
         mongoose.connection.collection(collectionName).insertOne(objectJSON, function (err, result) {
             if (err) {
-                console.log("Error, fail");
+                console.log(err);
             } else {
                 console.log("Successfully added into database!");
             }
         })
+      })
     },
 
     insertManyInDatabase: function (objectJSON, collectionName) {
+      return new Promise ((resolve, reject) => {
         mongoose.connection.collection(collectionName).insertMany(objectJSON, function (err, result) {
             if (err) {
-                console.log("Error, fail");
+                console.log(err);
             } else {
                 console.log("Successfully added into database!");
             }
         })
+      })
     },
 
-    createUser: function (objectJSON) {
-        mongoose.connection.collection(users).insertOne(objectJSON, function (err, result) {
+    createUser: function (objectJS) {
+      return new Promise((resolve, reject) => {
+        mongoose.connection.collection("users").insertOne(objectJS, function (err, result) {
             if (err) {
-                console.log("Error, fail");
+                reject(err);
             } else {
-                console.log("Successfully added into database!");
+                resolve("Successfully added into database!");
             }
         })
-    }
+      })
+    },
 
+    checkUserCredential: function (objectJSON) {
```

Comment:
This one should also use a promise rather than logging to the console when things succeed or fail. How does a successful login even affect the state of the system here?